lib/src/simple_op_store.rs

```diff
@@ -84,88 +203,146 @@ fn not_found_to_store_error(err: std::io::Error) -> OpStoreError {
 
 impl OpStore for SimpleOpStore {
     fn read_view(&self, id: &ViewId) -> OpStoreResult<View> {
-        let path = self.view_path(id);
-        let mut file = File::open(path).map_err(not_found_to_store_error)?;
+        self.delegate.read_view(id)
+    }
 
-        let proto: crate::protos::op_store::View = Message::parse_from_reader(&mut file)?;
-        Ok(view_from_proto(&proto))
+    fn write_view(&self, view: &View) -> OpStoreResult<ViewId> {
+        self.delegate.write_view(view)
+    }
+
+    fn read_operation(&self, id: &OperationId) -> OpStoreResult<Operation> {
+        self.delegate.read_operation(id)
+    }
+
+    fn write_operation(&self, operation: &Operation) -> OpStoreResult<OperationId> {
+        self.delegate.write_operation(operation)
+    }
+}
+
+#[derive(Debug)]
+struct ThriftOpStore {
+    path: PathBuf,
+}
+
+impl ThriftOpStore {
+    fn init(store_path: PathBuf) -> Self {
+        fs::create_dir(store_path.join("views")).unwrap();
+        fs::create_dir(store_path.join("operations")).unwrap();
+        Self::load(store_path)
+    }
+
+    fn load(store_path: PathBuf) -> Self {
+        ThriftOpStore { path: store_path }
+    }
+
+    fn view_path(&self, id: &ViewId) -> PathBuf {
+        self.path.join("views").join(id.hex())
+    }
+
+    fn operation_path(&self, id: &OperationId) -> PathBuf {
+        self.path.join("operations").join(id.hex())
+    }
+}
+
+impl OpStore for ThriftOpStore {
+    fn read_view(&self, id: &ViewId) -> OpStoreResult<View> {
+        let path = self.view_path(id);
+        let file = File::open(path).map_err(not_found_to_store_error)?;
+        let mut protocol = TCompactInputProtocol::new(&file);
+        let thrift_view =
+            simple_op_store_model::View::read_from_in_protocol(&mut protocol).unwrap();
+        Ok(view_from_thrift(&thrift_view))
     }
 
     fn write_view(&self, view: &View) -> OpStoreResult<ViewId> {
         let temp_file = NamedTempFile::new_in(&self.path)?;
 
-        let proto = view_to_proto(view);
-        let mut proto_bytes: Vec<u8> = Vec::new();
-        proto.write_to_writer(&mut proto_bytes)?;
+        let thrift_view = view_to_thrift(view);
+        let mut thrift_bytes: Vec<u8> = Vec::new();
+        let mut protocol = TCompactOutputProtocol::new(&mut thrift_bytes);
+        thrift_view.write_to_out_protocol(&mut protocol)?;
```

Comment:
This pattern, and its inverse, seems to occur in several places. Could we factor out a `{read,write}_thrift` helper function?

Reply:
Could still have a helper that we pass `T::write_to_out_protocol` into as an argument. Alternatively I could have a go at fixing the nix build.